packages/connectivity/src/scp-cf/destination/destination-service-types.ts

```diff
@@ -238,14 +238,44 @@ export type DestinationRetrievalOptions = CachingOptions & {
   retry?: boolean;
 };
 
+/**
+ * Destination for HTTP request where the URL is mandatory.
+ */
+export type HttpDestination = Destination & { url: string };
+
+/**
+ * Assertion that the Destination is a HttpDestination.
+ * This method comes in handy when you retrieved a destination from the destination service and need to check if it is a HttpDestination.
+ * @param destination - Destination or HttpDestination.
+ */
+export function assertHttpDestination(
+  destination: Destination | HttpDestination
+): asserts destination is HttpDestination {
+  if (!isHttpDestination(destination)) {
+    if (!destination.url) {
+      throw new Error(
+        `The 'url' property is not set for destination ${destination.name} which is mandatory if you use it as an 'HTTP destination`
+      );
+    }
+    throw new Error(
+      `The 'type' property is  ${destination.type} instead of  HTTP for destination '${destination.name}' which is mandatory if you use it as an 'HTTP destination`
+    );
+  }
+}
+
 /**
  * Type guard to find if object is a Destination.
- * @param destination - Destination to be checked
- * @returns boolean
- * @internal
+ * @param destination - Destination to be checked.
+ * @returns Boolean.
  */
-export function isDestination(destination: any): destination is Destination {
-  return destination.url !== undefined;
+export function isHttpDestination(
+  destination: HttpDestination | Destination | null
+): destination is HttpDestination {
+  return (
+    !!destination &&
+    !!destination.url &&
```

Comment:
[q] Could this return an incorrect result, e.g. if the `destination.url` is an empty string?
```
{
  url: "",
  type: "HTTP"
}
```

Reply:
[q] Previously it looked like we also supported `Type` (with capital T). If this changed, it should be mentioned in the compatibility note.

Reply:
This should be the same. I think in the past we did already an assertion before the transformation of the destination. Not it is done later on when the `type` property is already in lower case.